flink-libraries/flink-table/src/test/java/org/apache/flink/table/api/java/stream/sql/ProcTimeRowStreamAggregationSqlITCase.java

```diff
@@ -0,0 +1,317 @@
+		expected.add("1,0,0"); 
+		expected.add("2,1,1"); 
+		expected.add("2,3,1"); 
+		expected.add("3,3,3"); 
+		expected.add("3,7,3"); 
+		expected.add("3,9,4"); 
+		expected.add("4,6,6"); 
+		expected.add("4,13,6"); 
+		expected.add("4,15,7"); 
+		expected.add("4,17,8"); 
+		expected.add("5,10,10");
+		expected.add("5,21,10");
+		expected.add("5,23,11");
+		expected.add("5,26,12");
+		expected.add("5,27,13");
+
+		StreamITCase.compareWithList(expected);
+	}
+	
+	@Test
+	public void testGlobalSumAggregatation() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		StreamTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(env);
+		StreamITCase.clear();
+
+		env.setParallelism(1);
+		
+		DataStream<Tuple5<Integer, Long, Integer, String, Long>> ds = StreamTestData.get5TupleDataStream(env);
+		Table in = tableEnv.fromDataStream(ds, "a,b,c,d,e");
+		tableEnv.registerTable("MyTable", in);
+
+		String sqlQuery = "SELECT SUM(c) OVER (ORDER BY procTime() ROWS BETWEEN 2 PRECEDING AND CURRENT ROW) AS sumC FROM MyTable";
+		Table result = tableEnv.sql(sqlQuery);
+
+		DataStream<Row> resultSet = tableEnv.toDataStream(result, Row.class);
+		resultSet.addSink(new StreamITCase.StringSink());
+		env.execute();
+
+		List<String> expected = new ArrayList<>();
+		expected.add("0");
+		expected.add("1");
+		expected.add("3");
+		expected.add("5");
+		expected.add("7");
+		expected.add("9");
+		expected.add("11");
+		expected.add("13");
+		expected.add("15");
+		expected.add("17");
+		expected.add("19");
+		expected.add("21");
+		expected.add("23");
+		expected.add("26");
+		expected.add("27");
+
+		StreamITCase.compareWithList(expected);
+	}
+	
+	@Test
+	public void testAvgAggregatation() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		StreamTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(env);
+		StreamITCase.clear();
+
+		env.setParallelism(1);
+		
+		DataStream<Tuple5<Integer, Long, Integer, String, Long>> ds = StreamTestData.get5TupleDataStream(env);
+		Table in = tableEnv.fromDataStream(ds, "a,b,c,d,e");
+		tableEnv.registerTable("MyTable", in);
+
+		String sqlQuery = "SELECT a, AVG(c) OVER (PARTITION BY a ORDER BY procTime() ROWS BETWEEN 2 PRECEDING AND CURRENT ROW) AS avgC, e FROM MyTable";
+		Table result = tableEnv.sql(sqlQuery);
+
+		DataStream<Row> resultSet = tableEnv.toDataStream(result, Row.class);
+		resultSet.addSink(new StreamITCase.StringSink());
+		env.execute();
+
+		List<String> expected = new ArrayList<>();
+		expected.add("1,0,1");
+		expected.add("2,1,1");
+		expected.add("2,1,2");
+		expected.add("3,3,2");
+		expected.add("3,3,2");
+		expected.add("3,4,3");
+		expected.add("4,6,1");
+		expected.add("4,6,2");
+		expected.add("4,7,1");
+		expected.add("4,8,2");
+		expected.add("5,10,1");
+		expected.add("5,10,3");
+		expected.add("5,11,3");
+		expected.add("5,13,2");
+		expected.add("5,13,2");
```

Comment:
This result incorrect.The reason for the error I left the comments in `DataStreamOverAggregate.scala`